### ISSUE TITLE
changes to slides

### ISSUE THREAD

**Issue by surbhigoel77** [Cambridge-ICCS/practical-ml-with-pytorch#61]
Surbhi and I plan to spend more time explaining the theory of NNs with more focus on: 

- Datasets. Dimensions. 
- An expanded introduction: 
  - Potential applications. What NNs can do...
  - Other alternatives to NNs. 
  - More of an overview of pytorch. What is jax etc. 
- Links to previous statistical techniques like regression. 

### Specific notes on current slides: 

11/29 
- neural is spelt nerual
- some weird aberration on the slide
- weird formatting here. 

13/29 
- "workshop lecture thing"

### General notes on slides
- quite a brief overview of nns
        - do mention 3blue1brown ml course, can't do much better than this.
        - could we include some of this content in additional slides to better visualise some concepts? 
- jumps straight into SGD without much intro
	- more of a justification for why we're using NNs
- more images
	- visualising data 
		- tabulated data
	- potentials of what NNs can do  
- comparison of different models for different datasets.
  - Other methods like SVMs instead of NNs. 
- Drawbacks of NNs? 
         - black-boxes. https://towardsdatascience.com/the-math-behind-kan-kolmogorov-arnold-networks-7c12a164ba95
- pytorch vs scikit learn
  - why we even need pytorch

### Summary of what is currently covered

Penguin classification:
- loading penguin dataframe and inspecting data
- introduce a `torch.utils.data.dataset`. 
- split into train and validation
- transforming the data using `torchvision.transforms.Compose`

### Suggestions for guided tutorial / workshp 
- Add some boilerplate / template for them to add to. Notes. 
- Extensions. Compare different methods?

### Other resources
- Stanford ML resources

### Todos
- Think about what content we want to include as part of the expanded introduction. 
   - [ ] Look into Quarto 
   - [ ] How to improve slides, instead of starting with SGD. 
   - [ ] Might be helpful to look at other resources here. https://github.com/phlippe/uvadlc_notebooks
- Notebook improvements. 
  - [ ] Adding more content to the exercises to make them easier to complete
  - [ ] Statistical summary of data. max. / min / missing values ...
  - [ ]

**Comment by surbhigoel77** [comment]
Started with below slides:

1. What is Machine Learning
2. What are Neural Networks

**Comment by surbhigoel77** [comment]
@ma595 you can mention the slides that you change here

**Comment by ma595** [comment]
@jdenholm and I discussed this briefly last week. I'll attempt to summarise when I have some time. If @jdenholm can write a sentence summarising what he mentioned that would be great, but I appreciate that dedicating time to this might not be easy at the moment.

**Comment by jdenholm** [comment]
I think I suggested using PyTorch to fit a straight line, and then more general linear regression, as an intro to gradient descent and how it's useful for fitting functions.
